### DISCUSSION TITLE
Can we output libraries instead of websites ?

### DISCUSSION THREAD

**Discussion by ahna92** [varabyte/kobweb#581] [Q&A] [answered]
Can kobweb output a node js library for example, that can be integrated seamlessly with react website ? 
so the library will be like basic components 

Understand the complexity , just checking :D

**Comment by bitspittle** [comment, marked as answer]
Definitely Kobweb as a whole is designed to generate websites. However, there is its Silk widget library that is part of it which you can use on its own, maybe that can be imported? Probably not 

Note that Compose HTML (and by extension Kobweb) exist because people *don't* want to use React/JS, so compatibility with it is not really something I think much about at all.

You might also want to check the [Kotlin-bootstrap](https://github.com/stevdza-san/KotlinBootstrap) project. That might be more relevant here.

**Reply by ahna92** [reply]
Got it 

The reason for this question is 
If we have a design system built with compose 
Exporting into web would sync the 2 platform 

And avoid migrating existing projects (built with react) 

It’s related to tech adoption 
Start small

**Reply by bitspittle** [reply]
Sounds like you just need to break your project up into separate modules? You can create Kobweb libraries, maybe that's what you're asking?

https://github.com/varabyte/kobweb#splitting-kobweb-code-across-multiple-modules
